libraries/AP_Frsky_Telem/AP_Frsky_Telem.cpp

```diff
@@ -668,13 +666,18 @@ uint32_t AP_Frsky_Telem::calc_ap_status(void)
  */
 uint32_t AP_Frsky_Telem::calc_home(void)
 {
-    uint32_t home;
-
-    // distance between vehicle and home location in meters
-    home = prep_number(roundf(*_ap.home_distance * 0.01f), 3, 2);
-    // altitude between vehicle and home location in decimeters
+    uint32_t home = 0;
     Location loc;
-    if (_ahrs.get_position(loc)) {
+    if (_ahrs.get_position(loc)) {            
+        // check home_loc is valid
+        Location home_loc = _ahrs.get_home();
```

Comment:
you should use "const Location &home_loc" here, to avoid copying the structure and allocating unnecessary stack space